Lib/defconQt/glyphView.py

```diff
@@ -217,34 +333,200 @@ def __init__(self, glyph, parent=None):
+                return
+            color = qcolor.getRgbF()
+        # will trigger Layer.ColorChanged
+        self.view.layer.color = color
+
+    def _layerColorChange(self, notification):
+        layer = notification.object
+        if layer is self.view.layer:
+            self._setlayerColorButtonColor()
+
+    def _setlayerColorButtonColor(self):
+        color = self.view.getLayerColor()
+        if color is None:
+            icon = QIcon(":/resources/defaultColor.svg")
+        else:
+            # set the layer color to the button
+            pixmap = QPixmap(100, 100)
+            pixmap.fill(color)
+            icon = QIcon(pixmap)
+        self._layerColorButton.setIcon(icon)
+
+    def _updateComboBox(self):
+        comboBox = self._currentLayerBox
+        comboBox.blockSignals(True)
+        comboBox.clear()
+        for layer in self._layerSet:
+            comboBox.addItem(layer.name, layer)
+            if layer == self.view.layer:
+                comboBox.setCurrentText(layer.name)
+                continue
+        comboBox.addItem("New layer...", None)
+        comboBox.blockSignals(False)
+
+    def _setComboboxIndex(self, index):
+        comboBox = self._currentLayerBox
+        comboBox.blockSignals(True)
+        comboBox.setCurrentIndex(index)
+        comboBox.blockSignals(False)
+
+    def _makeNewLayerViaDialog(self):
+        newLayerName, ok = AddLayerDialog.getNewLayerName(self)
+        if ok:
+            # this should cause self._layerSetLayerAdded to be executed
+            self._layerSet.newLayer(newLayerName)
+            return self._layerSet[newLayerName]
+        else:
+            return None
+
+    def _changeLayerHandler(self, newLayerIndex):
+        comboBox = self._currentLayerBox
+        layer = comboBox.itemData(newLayerIndex)
+        if layer is None:
+            layer = self._makeNewLayerViaDialog()
+            if layer is None:
+                # restore comboBox to active index
+                index = self._layerSet.layerOrder.index(self.view.layer.name)
+                self._setComboboxIndex(index)
+                return
+
+        self.view.changeCurrentLayer(layer)
+        self._updateComboBox()
+        self._setlayerColorButtonColor()
+
+        app = QApplication.instance()
+        # setting the layer-glyph here
+        app.setCurrentGlyph(self.view._glyph)
+
     def changeGlyph(self):
         glyph = self.view._glyph
         newGlyph, ok = GotoDialog.getNewGlyph(self, glyph)
         if ok and newGlyph is not None:
-            self.view.setGlyph(newGlyph)
+            self._changeGlyph(newGlyph)
+            self._updateComboBox()
+
+    def _displayChanged(self, *args):
+        # redraw the view
+        self.view.redraw()
 
     def event(self, event):
         if event.type() == QEvent.WindowActivate:
             app = QApplication.instance()
+            # FIXME! which layer is this then -- default?
             app.setCurrentGlyph(self.view._glyph)
```

Comment:
I think the question is rather whether the global application should receive the default glyph or the real current glyph element. I think it's good like it is, because there may not always be a default glyph. I will remove the comment.